diff --git a/hayagriva.d.ts b/hayagriva.d.ts
index c0b6117..689ef7e 100644
--- a/hayagriva.d.ts
+++ b/hayagriva.d.ts
@@ -52,11 +52,11 @@ export class Library {
 /**
 * @returns {Iterator<any>}
 */
-  iter(): Iterator<any>;
+  iter(): Iterator<Entry>;
 /**
 * @returns {Iterator<any>}
 */
-  keys(): Iterator<any>;
+  keys(): Iterator<string>;
 /**
 * @param {string} key
 * @returns {Entry | undefined}
@@ -77,9 +77,7 @@ export class Library {
   nth(n: number): Entry | undefined;
 }

-export type InitInput = RequestInfo | URL | Response | BufferSource | WebAssembly.Module;
-
-export interface InitOutput {
+interface InitOutput {
   readonly memory: WebAssembly.Memory;
   readonly from_biblatex_str: (a: number, b: number, c: number) => void;
   readonly __wbg_library_free: (a: number) => void;
@@ -104,24 +102,3 @@ export interface InitOutput {
   readonly __wbindgen_free: (a: number, b: number, c: number) => void;
   readonly __wbindgen_start: () => void;
 }
-
-export type SyncInitInput = BufferSource | WebAssembly.Module;
-/**
-* Instantiates the given `module`, which can either be bytes or
-* a precompiled `WebAssembly.Module`.
-*
-* @param {SyncInitInput} module
-*
-* @returns {InitOutput}
-*/
-export function initSync(module: SyncInitInput): InitOutput;
-
-/**
-* If `module_or_path` is {RequestInfo} or {URL}, makes a request and
-* for everything else, calls `WebAssembly.instantiate` directly.
-*
-* @param {InitInput | Promise<InitInput>} module_or_path
-*
-* @returns {Promise<InitOutput>}
-*/
-export default function __wbg_init (module_or_path?: InitInput | Promise<InitInput>): Promise<InitOutput>;
diff --git a/hayagriva.js b/hayagriva.js
index 44b5d41..c3f41d2 100644
--- a/hayagriva.js
+++ b/hayagriva.js
@@ -1,3 +1,5 @@
+import { fetch } from 'fetch-with-file-support';
+
 let wasm;

 const cachedTextDecoder = (typeof TextDecoder !== 'undefined' ? new TextDecoder('utf-8', { ignoreBOM: true, fatal: true }) : { decode: () => { throw Error('TextDecoder not available') } } );
@@ -415,8 +417,10 @@ async function __wbg_load(module, imports) {

             } catch (e) {
                 if (module.headers.get('Content-Type') != 'application/wasm') {
-                    console.warn("`WebAssembly.instantiateStreaming` failed because your server does not serve wasm with `application/wasm` MIME type. Falling back to `WebAssembly.instantiate` which is slower. Original error:\n", e);
-
+                    if (/^https?:/.test(module.url)) {
+                        console.warn("`WebAssembly.instantiateStreaming` failed because your server does not serve wasm with `application/wasm` MIME type. Falling back to `WebAssembly.instantiate` which is slower. Original error:\n", e);
+                    }
+                    // Otherwise assume its 'file:' or something. Continues below try/catch.
                 } else {
                     throw e;
                 }
@@ -547,5 +551,4 @@ async function __wbg_init(input) {
     return __wbg_finalize_init(instance, module);
 }

-export { initSync }
-export default __wbg_init;
+await __wbg_init();
diff --git a/package.json b/package.json
index 7c71f12..8cf8516 100644
--- a/package.json
+++ b/package.json
@@ -20,5 +20,9 @@
   "types": "hayagriva.d.ts",
   "sideEffects": [
     "./snippets/*"
-  ]
-}
\ No newline at end of file
+  ],
+  "type": "module",
+  "dependencies": {
+    "fetch-with-file-support": "2.0.0"
+  }
+}
